roborock/code_mappings.py

```diff
@@ -1,183 +1,228 @@
-        7: "manual_mode",
-        8: "charging",
-        9: "charging_problem",
-        10: "paused",
-        11: "spot_cleaning",
-        12: "error",
-        13: "shutting_down",
-        14: "updating",
-        15: "docking",
-        16: "going_to_target",
-        17: "zoned_cleaning",
-        18: "segment_cleaning",
-        22: "emptying_the_bin",  # on s7+, see #1189
-        23: "washing_the_mop",  # on a46, #1435
-        26: "going_to_wash_the_mop",  # on a46, #1435
-        100: "charging_complete",
-        101: "device_offline",
-    },
-)
-
-RoborockErrorCode = create_code_enum(
-    "RoborockErrorCode",
-    {
-        0: "none",
-        1: "lidar_blocked",
-        2: "bumper_stuck",
-        3: "wheels_suspended",
-        4: "cliff_sensor_error",
-        5: "main_brush_jammed",
-        6: "side_brush_jammed",
-        7: "wheels_jammed",
-        8: "robot_trapped",
-        9: "no_dustbin",
-        12: "low_battery",
-        13: "charging_error",
-        14: "battery_error",
-        15: "wall_sensor_dirty",
-        16: "robot_tilted",
-        17: "side_brush_error",
-        18: "fan_error",
-        21: "vertical_bumper_pressed",
-        22: "dock_locator_error",
-        23: "return_to_dock_fail",
-        24: "nogo_zone_detected",
-        27: "vibrarise_jammed",
-        28: "robot_on_carpet",
-        29: "filter_blocked",
-        30: "invisible_wall_detected",
-        31: "cannot_cross_carpet",
-        32: "internal_error",
-    },
-)
-
-RoborockFanPowerCode = create_code_enum(
-    "RoborockFanPowerCode",
-    {
-        105: "off",
-        101: "silent",
-        102: "balanced",
-        103: "turbo",
-        104: "max",
-        108: "max_plus",
-        106: "custom",
-    },
-)
-
-RoborockMopModeCode = create_code_enum(
-    "RoborockMopModeCode",
-    {
-        300: "standard",
-        301: "deep",
-        303: "deep_plus",
-        302: "custom",
-    },
-)
-
-RoborockMopIntensityCode = create_code_enum(
-    "RoborockMopIntensityCode",
-    {
-        200: "off",
-        201: "mild",
-        202: "moderate",
-        203: "intense",
-        204: "custom",
-    },
-)
-
-RoborockDockErrorCode = create_code_enum(
-    "RoborockDockErrorCode",
-    {
-        0: "ok",
-        38: "water empty",
-        39: "waste water tank full",
-    },
-)
-
-RoborockDockTypeCode = create_code_enum(
-    "RoborockDockTypeCode",
-    {0: "no_dock", 1: "unknown", 2: "unknown", 3: "empty_wash_fill_dock", 4: "unknown", 5: "auto_empty_dock_pure"},
-)
-
-RoborockDockDustCollectionModeCode = create_code_enum(
-    "RoborockDockDustCollectionModeCode",
-    {
-        0: "smart",
-        1: "light",
-        2: "balanced",
-        4: "max",
-    },
-)
-
-RoborockDockWashTowelModeCode = create_code_enum(
-    "RoborockDockWashTowelModeCode",
-    {
-        0: "light",
-        1: "balanced",
-        2: "deep",
-    },
-)
+class RoborockStateCode(RoborockEnum):
+    starting = 1
```

Comment:
Shouldn't enum names be upper case? We could create a function to return it lower

Reply:
What would that function be? We would just call it on top of each time we need it lowercase?

I was thinking about translations of the entities, it my head it made more sense to be lower case but it could be upper case

Reply:
Let me test this when I have the chance. My first impression was that this is non standard. But if it helps with some scenarios I can work with it

Reply:
I wish I pushed my changes so that you could see it/ use it. But basically I have a function that builds a vacuum mode specification based on the model str that determines which e has to use